test/integration/pages/auth.js

```diff
@@ -0,0 +1,167 @@
+
+    user.set('email_check_hash', null);
+    await user.save(null, { method: 'update' });
+    store = initState();
+    client = new ApiClient(API_HOST);
+    triggers = new ActionsTrigger(client, store.dispatch);
+  });
+
+  after(async () => {
+    await user.destroy();
+    await bookshelf.knex.raw('DELETE FROM users;');
+  });
+
+  it('Login component should work', async (done) => {
+    const testComponent = <Login onLoginUser={triggers.login} />;
+    const wrapper = mount(testComponent);
+    wrapper.find('#loginUsername').node.value = userAttrs.username;
+    wrapper.find('#loginPassword').node.value = userAttrs.password;
+    wrapper.find('form').simulate('submit');
+
+    setTimeout(() => {
+      expect(user.get('id'), 'to equal', store.getState().getIn(['current_user', 'id']));
+      done();
+    }, 200);
+  });
+
+  describe('Register component', () => {
+    it('availableUsername should work', (done) => {
+      const testComponent =
+              <Register
+      fields={
+        {
+          username: {},
+          password: {},
+          passwordRepeat: {},
+          email: {},
+          agree: {}
+        }
+      }
+      form={{
+        forceValidate: () => {},
+        isValid: () => {},
+        onValues: () => {},
+        onRegisterUser: () => {},
+        onShowRegisterForm: () => {}
+      }}
+      onRegisterUser={() => {}}
+      onShowRegisterForm={() => {}}
+        />;
+      const wrapper = mount(testComponent);
+      expect(wrapper.find('#username').node.value, 'to be empty');
+
+      wrapper.find('#registerFirstName').node.value = 'test_firstname';
+      wrapper.find('#registerLastName').node.value = 'test_lastname';
+      wrapper.find('#registerFirstName').simulate('change');
+      wrapper.find('#registerLastName').simulate('change');
+
+      setTimeout(() => {
+        expect(wrapper.find('#username').node.value, 'to be non-empty');
+        done();
+      }, 50);
+    });
+  });
+
+  describe('Wrapped Register component', () => {
+    let userAttrs, user;
+    const email = 'test@example.com';
+
+    before(async () => {
+      userAttrs = UserFactory.build();
+      user = await User.create(userAttrs.username, userAttrs.password, email);
+    });
+
+    after(async () => {
+      await user.destroy();
+    });
+
+    it('should check on email currently taken', (done) => {
+      const testComponent = (
+        <WrappedRegister
+          onRegisterUser={() => {}}
+          onShowRegisterForm={() => {}}
+          />
+      );
+      const wrapper = mount(testComponent);
+
+      wrapper.find('#registerEmail').simulate('change', { target: { value: email } });
+      wrapper.find('#registerForm').simulate('submit');
+
+      setTimeout(() => {
+        expect(wrapper.state().errors.email, 'to equal', 'Email is taken');
+
+        done();
+      }, 100);
+    });
+
+    it('Should call "registerUser" trigger with no validation error', (done) => {
```

Comment:
This test doesn't check if the "registerUser" trigger have been called, it checks the side effects of pressing "submit". I'll try to find a way to create a mock for triggers.